lib/services/ios-notification-service.ts

```diff
@@ -0,0 +1,34 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import Future = require("fibers/future");
+
+export class IOSNotificationService implements IiOSNotificationService {
+	constructor(private $iOSNotification: IiOSNotification,
+		private $errors: IErrors,
+		private $projectData: IProjectData) { }
+
+	public awaitNotification(npc: Mobile.INotificationProxyClient, notification: string, timeout: number): IFuture<string> {
+		let future = new Future<string>();
+
+		let timeoutObject = setTimeout(() => {
```

Comment:
`timeoutCookie` or `timeoutToken` or `timeoutID` are better names for this object.